utils/fixed_decimal/src/decimal.rs

```diff
@@ -552,6 +552,197 @@ impl FixedDecimal {
         self.check_invariants();
     }
 
+    /// Increments the digits by 1. if the digits are empty, it will add
+    /// an element with value 1 and assign the value of the upper magnitude to
+    /// the magnitude.
+    fn increment_abs_by_one(&mut self) {
+        for i in (0..self.digits.len()).rev() {
+            self.digits[i] += 1;
+            if self.digits[i] < 10 {
+                return;
+            }
+            self.digits[i] = 0;
+        }
+
+        // Still there is a carry, add `1` in the beginning of the digits.
+        self.digits.insert(0, 1);
+        self.magnitude = {
+            if self.digits.len() == 1 {
+                self.upper_magnitude
+            } else {
+                self.magnitude + 1
+            }
+        };
+
+        if self.upper_magnitude < self.magnitude {
+            self.upper_magnitude = self.magnitude;
+        }
+
+        self.remove_trailing_zeros();
+    }
+
+    /// Removes the trailing zeros in `self.digits`
+    fn remove_trailing_zeros(&mut self) {
```

Comment:
Nit: the name of this function sounds like it removes the trailing zeros from the FixedDecimal (which would involve changing lower_magnitude), but this isn't what it's doing: it is just restoring an invariant on `self.digits`. Better name suggestions:

- `remove_trailing_zeros_from_digits_list`
- `compress_digits`
- `restore_digits_invariants`